devops/infrastructure/kubernetes/index.ts

```diff
@@ -0,0 +1,128 @@
+import * as awsx from '@pulumi/awsx'
+import * as eks from '@pulumi/eks'
+import * as k8s from '@pulumi/kubernetes'
+import * as pulumi from '@pulumi/pulumi'
+
+const awsConfig = new pulumi.Config('aws')
+
+// Create a VPC for our cluster.
+const vpc = new awsx.ec2.Vpc('vpc', { numberOfAvailabilityZones: 2 })
+
+// Create an EKS cluster with the default configuration.
+const cluster = new eks.Cluster('eksctl-my-cluster', {
+  vpcId: vpc.id,
+  subnetIds: vpc.publicSubnetIds,
+  instanceType: 't2.micro',
+  providerCredentialOpts: {
+    profileName: awsConfig.get('profile'),
+  },
+})
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig
+
+// Create a repository
+const repo = new awsx.ecr.Repository('my-repo')
+
+// Build an image and publish it to our ECR repository.
+
+export const colossusImage = repo.buildAndPushImage({
```

Comment:
I think you mentioned it will be possible to specify an image that could be fetched from a remote repo?
Building locally is fine for now, but it would be worth trying to optimize the size of the built image.

Reply:
As discussed, we will be building locally and pushing. Will try to optimize the image.